src/foam/nanos/auth/AgentUserAuthService.java

```diff
@@ -100,6 +82,45 @@ public User actAs(X x, User entity) throws AuthenticationException {
     return user;
   }
 
+  public boolean canActAs(X x, User agent, User entity) {
```

Comment:
Is there a better noun than "entity" we can use?  entity is very general, perhaps a name like "requestedUser" or "targetUser"

Reply:
"agency" seems reasonable given the constraints.

Reply:
"puppet" & "puppeteer" (joking)
"foregroundUser" & "backgroundUser" ?
"effectiveUser" & "ineffectiveUser"
"operativeUser" & "inoperativeUser"
"activeUser" & "inactiveUser"
"user" & "coUser"

Reply:
According to [this link](https://www.investopedia.com/terms/p/principal-agent-relationship.asp), **principal** might be the word we're looking for.